src/index.ts

```diff
@@ -103,6 +103,18 @@ export class PublicKey {
    * Pseudo-homomorphic Paillier multiplication
    */
   multiply(c: BigInteger.BigNumber, k: BigInteger.BigNumber) {
```

Comment:
It occurred to me that we can probably improve the type signatures for these:

`multiply<T extends BigInteger.BigNumber>(c: T, k: T): T {`

and likewise for the other methods. Up to you if you want to do that though, I guess it's only useful if client projects are using ts.

Reply:
Doing this causes more issues, If anyone has a better suggestion I'd be happy to push a change.

```'BigInteger' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'BigNumber'```

Reply:
You might have to specify the subtype when calling `multiply`. E.g.
```ts
const encMul = keyPair.publicKey.multiply<BigInteger.BigNumber>(cipher, 0);
```